### PR TITLE
dissto(id) now ignores own ID from particle list

### PR DESCRIPTION
**Pull request by davidsean**
Fixes 
distto(id) always returns a distance of 0,
this was due to checking ID with all others (including ID), thus always zero minimum distance.

Description of changes:
 - found a bug, fixed it.
 - doc string describes a bit better what it does

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

File: src/python/espressomd/analyze.pyx [hunk 1]
```diff
@@ -139,13 +139,16 @@ class Analysis(object):
         if id != None:
             if not isinstance(id, int):
                 raise ValueError("Id has to be an integer")
+            if id >= len(self._system.part):
```
Comment (KaiSzuttor) [inline src/python/espressomd/analyze.pyx]: That's not a good test... you can set a single particle with whatever id you want it to have

Comment (KaiSzuttor) [reply src/python/espressomd/analyze.pyx]: Better use sth like: if not id in system.part[:].id

Comment (davidsean) [reply src/python/espressomd/analyze.pyx]: good suggestion, I somehow didn't make the realization that the ID property and index were different.

**Comment by davidsean** [comment]
I am not sure why the ICP CUDA build failed

**Comment by KaiSzuttor** [comment]
Did you try to restart it by hand? Thats actually not possible

**Comment by davidsean** [comment]
no ( I wouldn't know how to), it's quite weird

**Comment by davidsean** [comment]
should I try a dummy commit?

**Comment by KaiSzuttor** [comment]
You can try but it's a general problem we have to deal with

**Review by fweik — APPROVED** [review]

**Pull request merged**